src/engine/renderer/gl_shader.h

```diff
@@ -3731,10 +3732,12 @@ class u_ColorModulateColorGen :
 				break;
 		}
 
-		if ( needAttrib ) {
-			_shader->AddVertexAttribBit( ATTR_COLOR );
-		} else {
-			_shader->DelVertexAttribBit( ATTR_COLOR );
+		if ( !needAttrib ) {
+			/* Originally, this controlled whether the vertex color array was used,
+			now it does the equivalent by setting the color in the shader in such way as if it was using
+			the default OpenGL values for the disabled arrays (0.0, 0.0, 0.0, 1.0)
+			This allows to skip the vertex format change */
```

Comment:
Is that related to "Vertex shader in program 11 is being recompiled based on GL state." warnings?